include/IECore/MeshAlgo.h

```diff
@@ -54,6 +54,8 @@ std::pair<PrimitiveVariable, PrimitiveVariable> calculateTangents( const MeshPri
 	const std::string &position = "P"
 );
 
+PrimitiveVariable resamplePrimitiveVariable( const MeshPrimitive *mesh, const std::string &primitiveVariable, PrimitiveVariable::Interpolation interpolation );
```

Comment:
I think it's quite clear from the return type & `const MeshPrimitive` the intent of the API. Sorry I should have added a comment also and will correct this.

I prefer the simpler interface of named primvars and raising exceptions if there it's missing or not the right interpolation. Pushing this onto the caller means more boiler plate code replicated in the call sites. Also resamplePrimitiveVariable is responsible for finding the PrimVar on the MeshPrimitive. A caller might get a primVar from another Primitive and attempt to use it in this function and we'd have to guard against this also.